cmd/revokeuser.go

```diff
@@ -110,33 +111,38 @@ func (p *RevokeUserParams) Load(ctx ActionCtx) error {
 		return err
 	}
 
-	if p.user == "" {
+	if p.user == "" && p.userPubKey == "" {
 		n := ctx.StoreCtx().DefaultUser(p.AccountContextParams.Name)
 		if n != nil {
 			p.user = *n
 		}
 	}
 
-	if p.user == "" {
-		return fmt.Errorf("user is required")
+	if p.userPubKey == "" && p.user == "" {
+		return fmt.Errorf("user or public-user-key are required")
+	} else if p.userPubKey != "" && p.user != "" {
+		return fmt.Errorf("user and public-user-key are mutually exclusive")
 	}
 
 	p.claim, err = ctx.StoreCtx().Store.ReadAccountClaim(p.AccountContextParams.Name)
 	if err != nil {
 		return err
 	}
+	if p.user != "" {
+		userClaim, err := ctx.StoreCtx().Store.ReadUserClaim(p.AccountContextParams.Name, p.user)
+		if err != nil {
+			return err
+		}
 
-	userClaim, err := ctx.StoreCtx().Store.ReadUserClaim(p.AccountContextParams.Name, p.user)
-	if err != nil {
-		return err
-	}
+		if err != nil || userClaim == nil {
+			return fmt.Errorf("user is required")
+		}
 
-	if err != nil || userClaim == nil {
-		return fmt.Errorf("user is required")
+		p.userPubKey = userClaim.Subject
+	} else if !nkeys.IsValidPublicUserKey(p.userPubKey) {
```

Comment:
Key validation should happen on the validate, that way the interactive doesn't have to validate as well

Reply:
fixed